Add API functionality

KDlamini opened the pull request:
This feature will be covering the following functionality:
- Create a new game with the Leaderboard API using the POST method.
- Implement the "Refresh" button for receiving data from the API.
- Implement the form "Submit" button for sending data to the API.
- Uses async and await JavaScript features to consume the API

KDlamini committed:
Update README and license.

KDlamini committed:
Update API functionality.
- Declare json variable and return it in the createGame api request.

dasileker commented on README.md:
```diff
@@ -1,2 +1,81 @@
+
+This project is part of the Microverse curriculum. The main objective is to understand how to consume APIs.
+
+## Live Preview
+[https://kdlamini.github.io/Leaderboard/](https://kdlamini.github.io/Leaderboard/)
```
- [ ] The `Live demo` is not working, please update your link.

dasileker commented on README.md:
````diff
@@ -1,2 +1,81 @@
+- Stylelint
+- Eslint
+
+## Getting Started
+
+To get a local copy of this project:
+
+Clone this repository or download the Zip folder:
+```
````
- [ ] The `Getting Started` should have more details on how the user can download the repo and navigate to it.

dasileker commented on src/api.js:
```diff
@@ -38,7 +38,8 @@ export const saveToApi = async (gameID, user, score) => {
     });
-    return await response.json();
+    const json = await response.json();
+    return json;
```
- [ ] I left this review here because I can't select the right line.

- [ ] The `createGame && the getScore` do not work, the **API** is not showing any entered data provided. I do see a lot of names and scores but when I try to add mine the API doesn't add anything???

dasileker reviewed (changes requested):
### **Hi @KDlamini,** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

KDlamini committed:
Update recentScores, addScore and api files.
- Update README with more details under 'getting started'.
- update dist folder.

KDlamini committed:
Update README.

KDlamini committed:
Refactor gameID function from returning a promise to return a string id.
- Set min-height for score table.

KDlamini committed:
Build app for deployement and update README live demo.

KDlamini committed:
Update validation for loading API when gameId is null.
- Add container for table for better dynamic display.
- run build to update dist folder.

KDlamini committed:
Fix linter errors.

KDlamini committed:
Add cloning instructions on README.

george-shammar reviewed (approved):
Hi @KDlamini , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

:heavy_check_mark: I was able to add score and the score reflected. :clap: 

Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
